### PR TITLE
refactor(uncrustify): change config to better align with style guide

### PR DESCRIPTION
**Pull request by dundargoc**
Change style rules:

- ~~align comments~~
- add space around `=`, `+=` ...
- other small stuff.

### CODE REVIEW

**Comment by justinmk** [comment]
Although aligning comments is nice, it needs to stay manual, to avoid diff churn.

File: src/nvim/eval.c [hunk 1]
```diff
@@ -151,103 +151,103 @@ static struct vimvar {
...
-  VV(VV_HLSEARCH,         "hlsearch",         VAR_NUMBER, 0),
-  VV(VV_OLDFILES,         "oldfiles",         VAR_LIST, 0),
-  VV(VV_WINDOWID,         "windowid",         VAR_NUMBER, VV_RO_SBX),
-  VV(VV_PROGPATH,         "progpath",         VAR_STRING, VV_RO),
-  VV(VV_COMPLETED_ITEM,   "completed_item",   VAR_DICT, VV_RO),
-  VV(VV_OPTION_NEW,       "option_new",       VAR_STRING, VV_RO),
-  VV(VV_OPTION_OLD,       "option_old",       VAR_STRING, VV_RO),
-  VV(VV_OPTION_OLDLOCAL,  "option_oldlocal",  VAR_STRING, VV_RO),
+  VV(VV_COUNT, "count", VAR_NUMBER, VV_RO),
+  VV(VV_COUNT1, "count1", VAR_NUMBER, VV_RO),
```
Comment (justinmk) [inline src/nvim/eval.c:155]: is this because the alignment was disabled? would prefer if crusty doesn't change this

Comment (justinmk) [reply src/nvim/eval.c:155]: there are many such blocks though

File: src/nvim/buffer_defs.h [hunk 2]
```diff
@@ -1512,6 +1512,6 @@ static inline int win_hl_attr(win_T *wp, int hlf)
 
 /// Macros defined in Vim, but not in Neovim
 #define CHANGEDTICK(buf) \
-  (=== Include buffer.h & use buf_(get|set|inc) _changedtick ===)
+  (=== Include buffer.h & use buf_(get | set | inc) _changedtick ===)
```
Comment (bfredl) [inline src/nvim/buffer_defs.h:1515]: we could add quotes `"` around the text to avoid the mangling

File: src/nvim/edit.c [hunk 3]
```diff
@@ -7989,7 +7989,7 @@ static bool ins_esc(long *count, int cmdchar, bool nomove)
   may_trigger_modechanged();
   // need to position cursor again when on a TAB
   if (gchar_cursor() == TAB) {
-    curwin->w_valid &= ~(VALID_WROW|VALID_WCOL|VALID_VIRTCOL);
+    curwin->w_valid &= ~(VALID_WROW | VALID_WCOL | VALID_VIRTCOL);
```
Comment (bfredl) [inline src/nvim/edit.c:7992]: could we change `|` to don't have space? think the old looked better.

Comment (dundargoc) [reply src/nvim/edit.c:7992]: No, not that I'm aware of.

File: src/nvim/event/process.c [hunk 4]
```diff
@@ -292,7 +292,7 @@ static void decref(Process *proc)
   }
 
   Loop *loop = proc->loop;
-  kliter_t(WatcherPtr) **node = NULL;
+  kliter_t(WatcherPtr) * *node = NULL;
```
Comment (zeertzjq) [inline src/nvim/event/process.c:295]: It seem uncrustify cannot recognize `kliter_t(WatcherPtr)` as a type.

**Review by justinmk — APPROVED** [review]

**Comment by zeertzjq** [comment]
I guess this needs a rebase?

**Comment by dundargoc** [comment]
@zeertzjq not needed.

File: src/nvim/viml/parser/expressions.c [hunk 5]
```diff
@@ -2646,22 +2646,17 @@ ExprAST viml_pexpr_parse(ParserState *const pstate, const int flags)
...
+          ADD_IDENT(NEW_NODE_WITH_CUR_POS(cur_node,
+                                          kExprNodeCurlyBracesIdentifier);
+                    cur_node->data.fig.opening_hl_idx = kv_size(*pstate->colors);
+                    cur_node->data.fig.type_guesses.allow_lambda = false;
+                    cur_node->data.fig.type_guesses.allow_dict = false;
+                    cur_node->data.fig.type_guesses.allow_ident = true;
+                    kvi_push(ast_stack, &cur_node->children);
+                    if (pt_is_assignment(cur_pt)) {
+            kvi_push(pt_stack, kEPTExpr);
+          }
```
Comment (zeertzjq) [inline src/nvim/viml/parser/expressions.c:2658]: Hmm, the indent here doesn't look right.

**Review by justinmk — APPROVED** [review]

**Pull request merged**